docs/DeveloperGuide.md

```diff
@@ -2,28 +2,454 @@
+    - `INDEX`: the index number of the statement in the financial report, stored as an int
+- Commands in the correct format will then be parsed to extract index, and a `DeleteCommand` object will
+  be created with the relevant attribute information
+- `RainyDay` will then call `execute` method in `Command`, where the indicated transaction will be deleted from the
+  financial report
+
+The sequence diagram for the implementation of delete is as shown below:
+
+![DeleteCommandSequenceDiagram.png](images\DeveloperGuide\DeleteCommandSequenceDiagram.png)
+
+#### Design considerations
+
+Format of delete command
+
+- Alternative 1 (current choice): usage of index to identify the statement to be deleted
+    - Pros: able to identify the correct statement to be deleted following the index assigned to the statement
+    - Cons: may not be intuitive for the user, as they have to look up the index of the statement before deletion
+- Alternative 2: usage of full statement information to identify the statement to be deleted
+    - Pros: more intuitive for the user
+    - Cons: user needs to know the full details of the statement and needs to type the full statement information for
+      identifying the statement to be deleted
+
+### Implementation of regex and parser
+
+The parser class is used to extract the necessary information from the user's input. Our aim is to make the parser as
+user-friendly and flexible as possible, and at the same time also reducing the margin for error and bugs. Thus, we
+decided to implement using regex and use the features provided by the libraries java.util.regex.Matcher and
+java.util.regex.Pattern.
+
+The first word of the user's input is checked and will be matched to one of the functions in the following list:
+
+- addStatement(String userInput): Parses an 'add' instruction from a user
+- parseDeleteStatement(String userInput): Parses a 'delete' instruction from a user
+
+#### addStatement(String userInput)
+
+This makes up the bulk of the class, as there are many fields for the add instruction, thus resulting in many
+variations of inputs. The following shows the format of input we expect from a user:
+
+Adds a new transaction to the financial report.
+
+Format: `add -DIRECTION DESCRIPTION $AMOUNT -c CATEGORY -date DD/MM/YYYY`
+
+* The `DIRECTION` to be `in` signifying an inflow type of transaction, or `out` signifying an outflow type of
+  transaction. This is a required field
+* `DESCRIPTION` is a required field.
+* The `$AMOUNT` takes in a number, is also a required field.
+* `-c CATEGORY` is an optional field that takes in a user-defined category of the product
+* `-date DD/MM/YYYY` is an optional field that takes in the date of a transaction
+
+As this is quite a long command to parse, we will use regular expressions in the following steps to match and break
+down the instructions.
+
+1. function addStatement will call function returnRemainingInformation which will check whether the input contains the
```

Comment:
Consider making `addStatement` and `returnRemainingInformation` formatted as inline code so it looks clearer to the reader